File: tony-java-monolithic-demo/tony-dto/src/main/java/tony/mono/dto/req/UserLoginReq.java
package tony.mono.dto.req;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;


/**
 * 用户登录请求
 *
 * @author tangli
 * @date 2025/03/23 22:26
 */
@Schema(description = "登录请求")
@Getter
@Setter
public class UserLoginReq {

    @Schema(description = "用户名", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "请输入用户名")
    private String userName;

    @Schema(description = "密码", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "请输入密码")
    private String pwd;
}

File: tony-monolithic-demo/tony-dto/src/main/kotlin/com/tony/dto/req/UserLoginReq.kt
package com.tony.dto.req

import io.swagger.v3.oas.annotations.media.Schema
import javax.validation.constraints.NotBlank

/**
 *
 * @author tangli
 * @since 2020-11-03 14:40
 */
@Schema(description = "登录请求")
data class UserLoginReq(

    @Schema(description = "用户名", required = true)
    @get:NotBlank(message = "请输入用户名")
    val userName: String?,

    @Schema(description = "密码", required = true)
    @get:NotBlank(message = "请输入密码")
    val pwd: String?,
)
